extensions/git/src/commands.ts

```diff
@@ -1377,41 +1403,27 @@ export class CommandCenter {
 
 
 	@command('git.branch', { repository: true })
-	async branch(repository: Repository): Promise<void> {
-		const config = workspace.getConfiguration('git');
-		const branchValidationRegex = config.get<string>('branchValidationRegex')!;
-		const branchWhitespaceChar = config.get<string>('branchWhitespaceChar')!;
-		const validateName = new RegExp(branchValidationRegex);
-		const sanitize = (name: string) => {
-			name = name.trim();
-
-			if (!name) {
-				return name;
-			}
+	async branch(repository: Repository, ref?: string): Promise<void> {
+		const name = await inputNameForBranch();
 
-			return name.replace(/^\.|\/\.|\.\.|~|\^|:|\/$|\.lock$|\.lock\/|\\|\*|\s|^\s*$|\.$|\[|\]$/g, branchWhitespaceChar);
-		};
-
-		const result = await window.showInputBox({
-			placeHolder: localize('branch name', "Branch name"),
-			prompt: localize('provide branch name', "Please provide a branch name"),
-			ignoreFocusOut: true,
-			validateInput: (name: string) => {
-				if (validateName.test(sanitize(name))) {
-					return null;
-				}
+		if (!name) {
+			return;
+		}
 
-				return localize('branch name format invalid', "Branch name needs to match regex: {0}", branchValidationRegex);
-			}
-		});
+		await repository.branch(name, true, ref);
+	}
 
-		const name = sanitize(result || '');
+	@command('git.branchFrom', { repository: true })
+	async createBranchFrom(repository: Repository): Promise<void> {
+		const refs = repositoryRefsToItems(repository);
+		const placeHolder = localize('select a ref to create a new branch from', 'Select a ref to create a new branch from');
+		const choice = await window.showQuickPick(refs, { placeHolder });
 
-		if (!name) {
+		if (!choice) {
 			return;
```

Comment:
Saw that in `branch()` this was recently changed to throw instead of return if the condition was false.
Should change here as well?

Reply:
It's OK. :+1: